Repeatedly deriving parse failure errors for 'user_huddle_changed' events

Other events also generate errors if they are not supported, but the `user_huddle_changed` event in particular was causing repeated errors, so I have raised it as an Issue. 

Error Log
```
[ERROR] [12/05/2022 13:38:18.675] [slack-akka.actor.default-dispatcher-22] [akka://slack/user/$a] [SlackRtmClient] Error reading event: {"type":"user_huddle_changed","user", ...
```

Stack Trace
```
play.api.libs.json.JsResultException: JsResultException(errors:List((,List(JsonValidationError(List(Invalid type property: {}),WrappedArray(user_huddle_changed))))))
        at play.api.libs.json.JsReadable.$anonfun$as$2(JsReadable.scala:25)
        at play.api.libs.json.JsError.fold(JsResult.scala:66)
        at play.api.libs.json.JsReadable.as(JsReadable.scala:24)
        at play.api.libs.json.JsReadable.as$(JsReadable.scala:23)
        at play.api.libs.json.JsObject.as(JsValue.scala:126)
        at slack.rtm.SlackRtmConnectionActor.$anonfun$onTextMessageReceive$1(SlackRtmClient.scala:186)
        at scala.util.Try$.apply(Try.scala:213)
        at slack.rtm.SlackRtmConnectionActor.onTextMessageReceive(SlackRtmClient.scala:186)
        at slack.rtm.SlackRtmConnectionActor$$anonfun$receive$1.applyOrElse(SlackRtmClient.scala:137)
        at akka.actor.Actor.aroundReceive(Actor.scala:537)
        at akka.actor.Actor.aroundReceive$(Actor.scala:535)
        at slack.rtm.SlackRtmConnectionActor.aroundReceive(SlackRtmClient.scala:116)
        at akka.actor.ActorCell.receiveMessage(ActorCell.scala:579)
        at akka.actor.ActorCell.invoke(ActorCell.scala:547)
        at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:270)
        at akka.dispatch.Mailbox.run(Mailbox.scala:231)
        at akka.dispatch.Mailbox.exec(Mailbox.scala:243)
        at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
        at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
        at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
        at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
        at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
```

--- a/models/src/main/scala/slack/models/Events.scala
+++ b/models/src/main/scala/slack/models/Events.scala
@@ -160,6 +160,10 @@ case class PrefChange(name: String, value: JsValue) extends SlackEvent
 
 case class UserChange(user: User) extends SlackEvent
 
+case class UserHuddleChanged(user: User) extends SlackEvent
+
+case class UserProfileChanged(user: User) extends SlackEvent
+
 case class TeamJoin(user: User) extends SlackEvent
 
 case class StarAdded(user: String,

--- a/models/src/main/scala/slack/models/package.scala
+++ b/models/src/main/scala/slack/models/package.scala
@@ -147,6 +147,8 @@ package object models {
   implicit val manualPresenceChangeFmt = Json.format[ManualPresenceChange]
   implicit val prefChangeFmt = Json.format[PrefChange]
   implicit val userChangeFmt = Json.format[UserChange]
+  implicit val userProfileChangeFmt = Json.format[UserProfileChanged]
+  implicit val userHuddleChangeFmt = Json.format[UserHuddleChanged]
   implicit val teamJoinFmt = Json.format[TeamJoin]
   implicit val starAddedFmt = Json.format[StarAdded]
   implicit val starRemovedFmt = Json.format[StarRemoved]
@@ -232,6 +234,8 @@ package object models {
         case e: ManualPresenceChange => Json.toJson(e)
         case e: PrefChange => Json.toJson(e)
         case e: UserChange => Json.toJson(e)
+        case e: UserProfileChanged => Json.toJson(e)
+        case e: UserHuddleChanged => Json.toJson(e)
         case e: TeamJoin => Json.toJson(e)
         case e: StarAdded => Json.toJson(e)
         case e: StarRemoved => Json.toJson(e)
@@ -322,6 +326,8 @@ package object models {
           case "manual_presence_change" => JsSuccess(jsValue.as[ManualPresenceChange])
           case "pref_change" => JsSuccess(jsValue.as[PrefChange])
           case "user_change" => JsSuccess(jsValue.as[UserChange])
+          case "user_profile_changed" => JsSuccess(jsValue.as[UserProfileChanged])
+          case "user_huddle_changed" => JsSuccess(jsValue.as[UserHuddleChanged])
           case "team_join" => JsSuccess(jsValue.as[TeamJoin])
           case "star_added" => JsSuccess(jsValue.as[StarAdded])
           case "star_removed" => JsSuccess(jsValue.as[StarRemoved])

--- a/src/test/resources/json-messages/user_change.json
+++ b/src/test/resources/json-messages/user_change.json
@@ -1,0 +1,32 @@
+{
+  "type": "user_change",
+  "user": {
+    "id": "U2BAA14A1B5",
+    "team_id": "T3C500576",
+    "name": "Some Name",
+    "color": "e85d72",
+    "is_bot": false,
+    "is_app_user": false,
+    "profile": {
+      "avatar_hash": "g95160e65b38",
+      "image_24": "https://example.com",
+      "image_32": "https://example.com",
+      "image_48": "https://example.com",
+      "image_72": "https://example.com",
+      "image_192": "https://example.com",
+      "image_512": "https://example.com",
+      "first_name": "Some Name",
+      "last_name": "Some Name",
+      "real_name": "Some Name",
+      "display_name": "Some Name",
+      "real_name_normalized": "Some Name",
+      "display_name_normalized": "Some Name",
+      "email": "noone@example.com",
+      "team": "T3C500576"
+    },
+    "updated": 1671223161,
+    "who_can_share_contact_card": "E418B758"
+  },
+  "cache_ts": 1671223161,
+  "event_ts": "1671223161.266300"
+}

--- a/src/test/resources/json-messages/user_huddle_changed.json
+++ b/src/test/resources/json-messages/user_huddle_changed.json
@@ -1,0 +1,32 @@
+{
+  "type": "user_huddle_changed",
+  "user": {
+    "id": "U2BAA14A1B5",
+    "team_id": "T3C500576",
+    "name": "Some Name",
+    "color": "e85d72",
+    "is_bot": false,
+    "is_app_user": false,
+    "profile": {
+      "avatar_hash": "g12340a11a11",
+      "image_24": "https://example.com",
+      "image_32": "https://example.com",
+      "image_48": "https://example.com",
+      "image_72": "https://example.com",
+      "image_192": "https://example.com",
+      "image_512": "https://example.com",
+      "first_name": "Some Name",
+      "last_name": "Some Name",
+      "real_name": "Some Name",
+      "display_name": "Some Name",
+      "real_name_normalized": "Some Name",
+      "display_name_normalized": "Some Name",
+      "email": "noone@example.com",
+      "team": "T3C500576"
+    },
+    "updated": 1671223161,
+    "who_can_share_contact_card": "E418B758"
+  },
+  "cache_ts": 1671223161,
+  "event_ts": "1671223161.266400"
+}

--- a/src/test/resources/json-messages/user_profile_changed.json
+++ b/src/test/resources/json-messages/user_profile_changed.json
@@ -1,0 +1,34 @@
+{
+  "type": "user_profile_changed",
+  "user": {
+    "id": "U61223C36",
+    "team_id": "T74EA48EB",
+    "name": "Some Name",
+    "color": "d1707d",
+    "is_bot": false,
+    "is_app_user": false,
+    "profile": {
+      "avatar_hash": "05656babab01",
+      "image_original": "https://example.com",
+      "image_24": "https://example.com",
+      "image_32": "https://example.com",
+      "image_48": "https://example.com",
+      "image_72": "https://example.com",
+      "image_192": "https://example.com",
+      "image_512": "https://example.com",
+      "image_1024": "https://example.com",
+      "first_name": "Some Name",
+      "last_name": "Some Name",
+      "real_name": "Some Name",
+      "display_name": "Some Name",
+      "real_name_normalized": "Some Name",
+      "display_name_normalized": "Some Name",
+      "email": "just.phil@example.com",
+      "team": "T74EA48EB"
+    },
+    "updated": 1671223135,
+    "who_can_share_contact_card": "E418B758"
+  },
+  "cache_ts": 1671223135,
+  "event_ts": "1671223135.322200"
+}

